addons/io/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/NeeoItemValueConverter.java

```diff
@@ -0,0 +1,170 @@
+
+    /** RegEx to identify format patterns. See java.util.Formatter#formatSpecifier (without the '%' at the very end). */
+    protected static final String IDENTIFY_FORMAT_PATTERN_PATTERN = "%(\\d+\\$)?([-#+ 0,(\\<]*)?(\\d+)?(\\.\\d+)?([tT])?([a-zA-Z])";
+
+    /** The service context */
+    private final ServiceContext context;
+
+    public NeeoItemValueConverter(ServiceContext context) {
+        this.context = context;
+    }
+
+    /**
+     * Converts the {@link State} from the given {@link NeeoDeviceChannel} to a {@link NeeoItemValue}. This method will
+     * convert enums to booleans and format string values to their associated pattern
+     *
+     * @param channel the non-null channel
+     * @param state the non-null state
+     * @return the non-null {@link NeeoItemValue}
+     */
+    public NeeoItemValue convert(NeeoDeviceChannel channel, State state) {
+        Objects.requireNonNull(channel, "channel cannot be null");
+        Objects.requireNonNull(state, "state cannot be null");
+
+        final String format = channel.getValue();
+        if (state instanceof OnOffType) {
+            return new NeeoItemValue(state == OnOffType.ON);
+        } else if (state instanceof OpenClosedType) {
+            return new NeeoItemValue(state == OpenClosedType.OPEN);
+        } else if (state instanceof UpDownType) {
+            return new NeeoItemValue(state == UpDownType.UP);
+        } else if (state instanceof DecimalType) {
+            if (StringUtils.isEmpty(format)) {
+                return new NeeoItemValue(((DecimalType) state).toBigDecimal());
+            }
+        } else if (state instanceof UnDefType) {
+            return new NeeoItemValue("-");
+        } else if (state instanceof StringType && channel.getType() == NeeoCapabilityType.IMAGEURL) {
+            return new NeeoItemValue(state.toString());
+        }
+
+        String itemValue;
+        if (StringUtils.isNotEmpty(format)) {
+            if (state == null || state instanceof UnDefType) {
+                itemValue = formatUndefined(format);
+            } else if (state instanceof Type) {
+                if (TransformationHelper.isTransform(format)) {
+                    try {
+                        final String transformed = TransformationHelper
+                                .transform(context.getComponentContext().getBundleContext(), format, state.toString());
+                        return new NeeoItemValue(transformed);
+                    } catch (NoClassDefFoundError ex) {
+                        // TransformationHelper is optional dependency, so ignore if class not found
+                    }
+                }
+
+                // The following exception handling has been added to work around a Java bug with formatting
+                // numbers. See http://bugs.sun.com/view_bug.do?bug_id=6476425
+                // Without this catch, the whole sitemap, or page can not be displayed!
```

Comment:
Stole that code from somewhere in openHAB (comments and all since it's talking about sitemaps/pages - which don't apply in io).  I removed the comment since it really doesn't apply - we want to catch formatting exceptions anyway...